### PR TITLE
#166980205 Setup circleci and coverage

### PR DESCRIPTION
**Pull request by ceexon**
### What does this PR do?
- Add CircleCI integration

### Description of Task to be completed?
- [x] Integrate CircleCI to the repo
- [x] Integrate code climate to the repo

### How should this be manually tested?
- After Pushing changes or creating a pull request, CircleCI is triggered
- Code climate and Circle CI badges are also updated

[CircleCI](https://circleci.com/gh/andela/ah-baqi-frontend) [Maintainability](https://codeclimate.com/github/andela/ah-baqi-frontend/maintainability) [Test Coverage](https://codeclimate.com/github/andela/ah-baqi-frontend/test_coverage)

### Any background context you want to provide?
- None

### What are the relevant pivotal tracker stories?
[#166980205](https://www.pivotaltracker.com/story/show/166980205)

### CODE REVIEW

**Comment by ceexon** [comment]
Circle CI is failing because it cannot detect the package.json file 
When merged it should work

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+  build:
+    docker:
+      - image: circleci/node:10.15.2
+    steps:
+      - checkout
+      - restore_cache: # special step to restore the dependency cache
+          key: dependency-cache-{{ checksum "package.json" }}
+      - run:
+          name: Setup dependencies
+          command: npm install
```
Comment (wekesa931) [inline .circleci/config.yml:12]: Consider using yarn package manager instead of npm as agreed by the team.

Comment (ceexon) [reply .circleci/config.yml:12]: I've not found a descriptive way of doing circle ci with yarn 
However, the only thing needed is the package.json and tests 
We can use npm to run tests on circle ci and use yarn in other places 

what do you think?

**Review by jmusila — APPROVED** [review]
Nice job

**Review by wangonya — APPROVED** [review]

**Comment by FlevianK** [comment]
Kindly fix houndci issues

**Pull request merged**